src/device/device.ts

```diff
@@ -203,21 +203,30 @@ export class MicrobitWebUSBConnection
     this.emit(EVENT_SERIAL_DATA, data);
   };
 
+  private flashing: boolean = false;
+  private disconnectAfterFlash: boolean = false;
   private visibilityReconnect: boolean = false;
   private visibilityChangeListener = () => {
     if (document.visibilityState === "visible") {
       if (
         this.visibilityReconnect &&
         this.status !== ConnectionStatus.CONNECTED
       ) {
+        this.disconnectAfterFlash = false;
         this.visibilityReconnect = false;
-        this.connect();
+        if (!this.flashing) {
+          this.connect();
+        }
```

Comment:
I think the flashing guard is redundant as we'll be connected if we're flashing.

Reply:
Though in some ways it makes it easier to reason about so I might leave it.